spark/src/main/scala/org/apache/comet/DataTypeSupport.scala

```diff
@@ -79,4 +79,14 @@ object DataTypeSupport {
     case _: StructType | _: ArrayType | _: MapType => true
     case _ => false
   }
+
+  def hasTemporalType(t: DataType): Boolean = t match {
+    case DataTypes.DateType | DataTypes.TimestampType | DataTypes.TimestampNTZType =>
```

Comment:
wondering if interval can also be considered as temporal? Probably not, it is more like duration rather than representing  any point of time